### ISSUE TITLE
Deserialize<T> does not exist?

### ISSUE THREAD

**Issue by TheColonel2688** [wojtpl2/ExtendedXmlSerializer#233]
In the readme, there is an example youing a  Deserialize<T>(reader) method. But this does not seem to exist.

![ctrls-sb-ben 2019-0130 12-28-00](https://user-images.githubusercontent.com/8753759/52000423-10a7d780-248b-11e9-9cb5-75082de89514.png)

**Comment by wojtpl2** [comment]
You have missing namespace.

śr., 30 sty 2019, 18:32: TheColonel2688 <notifications@github.com>
napisał(a):

**Comment by TheColonel2688** [comment]
Deserialize() exist, which namespace it deserializer<T>() in?

I have tried all of the ones in the nuget package
using ExtendedXmlSerializer.Configuration;
using ExtendedXmlSerializer.Core;
using ExtendedXmlSerializer.ReflectionModel;
using ExtendedXmlSerializer.ExtensionModel;
using ExtendedXmlSerializer.ContentModel;
using ExtendedXmlSerializer;

**Comment by Mike-EEE** [comment]
Hey @TheColonel2688 thanks for reporting this.  Do you know where in the readme you found this?  The `Deserialize<T>` is the extension method but I do not believe we have one that accepts a `reader` object.  The method that accepts the `reader` object is the non-generic method.  If you can point out where in the readme where this is wrong I will check it out and address it for you. 👍

**Comment by TheColonel2688** [comment]
Under the "XML Settings" section.
I now understand what is going on now. An extension method is being used to write to a string instead of a XmlWriter and then the string is being used to deserialize instead of a XmlReader.

You may what to change the line 
`var contents = serializer.Serialize(new XmlWriterSettings {Indent = true}, subject);`
to
`string contents = serializer.Serialize(new XmlWriterSettings {Indent = true}, subject);`

it may make it clearer to follow

**Comment by Mike-EEE** [comment]
Good point @TheColonel2688 ... I've created a new issue in #235 to track. 👍
